spec/rubocop/cop/style/map_into_array_spec.rb

```diff
@@ -0,0 +1,395 @@
+# frozen_string_literal: true
+
+RSpec.describe RuboCop::Cop::Style::MapIntoArray, :config do
+  it 'registers an offense and corrects when using `each` with `<<` to build an array' do
+    expect_offense(<<~RUBY)
+      dest = []
+      src.each { |e| dest << e * 2 }
+      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ Use `map` instead of `each` to map elements into an array.
+    RUBY
+
+    expect_correction(<<~RUBY)
+      src.map { |e| e * 2 }
+    RUBY
+  end
+
+  it 'registers an offense and corrects when using `each` with `push` to build an array' do
+    expect_offense(<<~RUBY)
+      dest = []
+      src.each { |e| dest.push(e * 2) }
+      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ Use `map` instead of `each` to map elements into an array.
+    RUBY
+
+    expect_correction(<<~RUBY)
+      src.map { |e| e * 2 }
+    RUBY
+  end
+
+  it 'registers an offense and corrects when using `each` with `append` to build an array' do
+    expect_offense(<<~RUBY)
+      dest = []
+      src.each { |e| dest.append(e * 2) }
+      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ Use `map` instead of `each` to map elements into an array.
+    RUBY
+
+    expect_correction(<<~RUBY)
+      src.map { |e| e * 2 }
+    RUBY
+  end
+
+  it 'registers an offense and corrects when a non-related operation precedes an `each` call' do
+    expect_offense(<<~RUBY)
+      dest = []
+      do_something
+      src.each { |e| dest << e * 2 }
+      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ Use `map` instead of `each` to map elements into an array.
+    RUBY
+
+    expect_correction(<<~RUBY)
+      do_something
+      src.map { |e| e * 2 }
```

Comment:
This autocorrect is extra tricky since if `do_something` is more code it might modify `dest`.